### PR TITLE
DS-3889 Tidy RemoveAt.ftable and add tests [revdep skip]

### PR DESCRIPTION
**Pull request by jrwishart**
- Improve removeArrayInputsBad code
- DS-3887 Add method for RemoveAt.ftable
- DS-3889 Add tests
- DS-3889 Add RemoveAt.ftable Rd [ci skip]
- DS-3889 Add ftable rds file for tests
- DS-3889 Improve existing tests [revdep skip]
- DS-3889 Update documentation [ci skip]
- DS-3889 Remove unused code [revdep skip]
- DS-3889 Simplify existing code [revdep skip]

### CODE REVIEW

File: R/removeat.R [hunk 1]
```diff
@@ -80,23 +80,19 @@ RemoveAt.matrix <- function(x, at = NULL, MARGIN = NULL, ignore.case = TRUE, spl
 #' @export
 RemoveAt.ftable <- function(x, at = NULL, MARGIN = NULL, ignore.case = TRUE, split = NULL)
```
Comment (mwmclean) [inline R/removeat.R:92]: Add `@noRd`, `@rdname`, or `@describeIn` so a separate Rd file isn't created.

File: R/removeat.R [hunk 2]
```diff
@@ -124,21 +134,11 @@ indicesToRetain <- function(names, at, length.x, ignore.case = TRUE, split = NUL
...
-        if (!is.null(split))
-            at <- ConvertCommaSeparatedStringToVector(at, split)
-        return(!names %in% at)
-    }
+        return(determineIndicesFromChar(at, names, ignore.case, split))
     # 'at' is numeric
     if (anyNA(at) || !AllIntegers(at))
         stop("'at' must contain character (string) or integer values.")
-    if (any(is.na(at) | at < 1))
+    if (any(at < 0))
```
Comment (mwmclean) [inline R/removeat.R]: Why change from `at < 1` to `at < 0`?

**Review by mwmclean — CHANGES_REQUESTED** [review]

File: tests/testthat/test-removeat.R [hunk 3]
```diff
@@ -198,3 +198,47 @@ test_that("RemoveAt: another vector and a list",
                         "NET Sugarred", "NET Sugarless", "NET"),
                 "Age in years"))
 })
+
+# Check flattened tables are filtered correctly
+
+ftbls <- readRDS("ftbls.rds")
```
Comment (mwmclean) [inline tests/testthat/test-removeat.R]: If the structure of ftables changes in a later version of R, your tests won't pick that up.

File: R/removeat.R (hunk 1 above)
Comment (jrwishart) [reply R/removeat.R:92]: This applies to all classes for this method as they are also missing it, so I'll do that to all.

File: R/removeat.R (hunk 2 above)
Comment (jrwishart) [reply R/removeat.R]: Typo, Fixed in [403ae35584b10b692a343945c3b34c18c89c38d1](https://github.com/Displayr/flipU/pull/5/commits/4f12c02931312da47279e8059ef2e2c3db257e84)  thanks

File: tests/testthat/test-removeat.R (hunk 3 above)
Comment (jrwishart) [reply tests/testthat/test-removeat.R]: Updated.

File: R/removeat.R [hunk 4]
```diff
@@ -1,7 +1,7 @@
 #' \code{RemoveAt}
 #'
 #' @description Removes one or more elements from an object. Dimensions are never
-#' If all elements are removed from a vector, then a 0-length vector is returned.
+#'              If all elements are removed from a vector, then a 0-length vector is returned.
```
Comment (mwmclean) [inline R/removeat.R:4]: Why would you intent this like this? Would be sweet if you could complete the sentence on the previous line or remove it.

**Review by mwmclean — CHANGES_REQUESTED** [review]

**Review by mwmclean — APPROVED** [review]

**Pull request merged**